### PR TITLE
[12.0][MIG] auth_signup_verify_email

### PR DESCRIPTION
**Pull request by Tardo**
PR superseded: https://github.com/OCA/server-auth/pull/69

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
What is the PR superseeded?

**Comment by Tardo** [comment]
@pedrobaeza PR updated with the requested info.

**Comment by pedrobaeza** [comment]
@ernestotejeda please review this PR

**Review by chienandalu — COMMENTED** [review]
I'm trying a functional test but I get this error

test-auth

I'd say the problem comes from this view: https://github.com/OCA/server-auth/pull/108/files#diff-add74761757ad34e815f5cd9f9933f97R9

File: auth_signup_verify_email/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+To configure this module, you need to:
+
+* `Properly configure your outgoing email server(s)
+  <https://www.odoo.com/es_ES/forum/help-1/question/how-to-configure-email-gateway-282#answer_290>`_.
```
Comment (yajo) [inline auth_signup_verify_email/readme/CONFIGURE.rst]:
```suggestion
  <https://www.odoo.com/forum/help-1/question/how-to-configure-email-gateway-282#answer_290>`_.
```

Let's use the :us: version.

File: auth_signup_verify_email/readme/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+* Install `validate_email <https://pypi.python.org/pypi/validate_email>`_
```
Comment (yajo) [inline auth_signup_verify_email/readme/INSTALL.rst]:
```suggestion
* Install `email_validator <https://pypi.org/project/email-validator/>`_
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
@chienandalu The problem is the SMTP configuration: https://github.com/OCA/server-auth/pull/108/files#diff-b76492f8c47aaeb881c43ddd50c764ccR67
'only_passwords' is used when you have a token, this token is sent via email to set the passwords.
Try using test@ñ0.com email, you can see that the form is sent correctly...

File: auth_signup_verify_email/readme/CONFIGURE.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+To configure this module, you need to:
+
+* `Properly configure your outgoing email server(s)
+  <https://www.odoo.com/forum/help-1/question/how-to-configure-email-gateway-282#answer_290>`_.
+* Go to *Settings > General Settings -> General settings*, search for
```
Comment (ernestotejeda) [inline auth_signup_verify_email/readme/CONFIGURE.rst:5]:
```suggestion
* Go to *Settings > General Settings*, search for
```

**Review by ernestotejeda — COMMENTED** [review]
Work great for me

- Generate README.rst again after make the change i've proposed to include all the changes made in the readme fragment files
- Squash to have a single migration commit

File: auth_signup_verify_email/readme/CONFIGURE.rst (hunk 3 above)
Comment (Tardo) [reply auth_signup_verify_email/readme/CONFIGURE.rst:5]: I think that is correct:
general_settings

Comment (Tardo) [reply auth_signup_verify_email/readme/CONFIGURE.rst:5]: Last changes are made by Yajo... it's correct squash?

**Review by chienandalu — COMMENTED** [review]
Sorry @Tardo But even with a right smtp config and my email I'm not able to signup on the in the instance.

test-auth-2

**Comment by Tardo** [comment]
@chienandalu can reproduce it on local?

**Comment by ernestotejeda** [comment]
It worked well for me on the runbot

**Review by victor-ditay — APPROVED** [review]
LGTM technical speaking. Waiting for functional approval of @chienandalu

**Comment by mohamedhagag** [comment]
it's working fine with me .

**Comment by pedrobaeza** [comment]
@Tardo please check the problem in Travis on the rebased branch.

**Comment by Tardo** [comment]
@pedrobaeza it's a incompatibility with 'auth_oauth' module.. these module doesn't modify the qcontext properly... https://github.com/odoo/odoo/blob/12.0/addons/auth_oauth/controllers/main.py#L116
I made a "hack" to prevent crashes.

**Comment by yajo** [comment]
> these module doesn't modify the qcontext properly

What's wrong there? :thinking:

**Comment by Tardo** [comment]
@Yajo the correct place to modify the qcontext is here: https://github.com/odoo/odoo/blob/11.0/addons/auth_signup/controllers/main.py#L101
If you change the xml and modify the qcontext in 'web_auth_signup' the view will broke if you don't use "super"...

File: auth_signup_verify_email/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+from odoo import _
+from odoo.http import request, route
+from odoo.addons.auth_signup.controllers.main import AuthSignupHome
+
+_logger = logging.getLogger(__name__)
+
+
+class SignupVerifyEmail(AuthSignupHome):
+
+    # HACK to prevent crashes if other modules modify "qcontext" in wrong
```
Comment (yajo) [inline auth_signup_verify_email/controllers/main.py]:
```suggestion
    # HACK https://github.com/odoo/odoo/pull/34690
```

And add it to roadmap please.

File: auth_signup_verify_email/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    # places
```
Comment (yajo) [inline auth_signup_verify_email/controllers/main.py]:
```suggestion
```

File: auth_signup_verify_email/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    def web_auth_signup(self, *args, **kw):
+        response = super(SignupVerifyEmail, self).web_auth_signup(*args, **kw)
+        if (request.params.get("login") and not
+                request.params.get("password")):
+            return self.passwordless_signup(request.params, response.qcontext)
+        else:
+            return response
+
+    def passwordless_signup(self, values, qcontext):
+        if qcontext['error'] == '--HACK--':
```
Comment (yajo) [inline auth_signup_verify_email/controllers/main.py:34]:
```suggestion
        # HACK https://github.com/odoo/odoo/pull/34690
        if qcontext['error'] == '--HACK--':
```

File: auth_signup_verify_email/readme/CONFIGURE.rst (hunk 3 above)
Comment (yajo) [reply auth_signup_verify_email/readme/CONFIGURE.rst:5]: For me it's OK, but thanks for asking.

File: auth_signup_verify_email/tests/test_verify_email.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+from mock import patch
+from lxml.html import document_fromstring
+from odoo.tests.common import at_install, post_install, HttpCase
+from odoo.addons.mail.models import mail_template
+from odoo.tools.misc import mute_logger
+
+
+@at_install(False)
+@post_install(True)
```
Comment (yajo) [inline auth_signup_verify_email/tests/test_verify_email.py]:
```suggestion
```

Avoid these things as much as possible.

File: auth_signup_verify_email/tests/test_verify_email.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from mock import patch
+from lxml.html import document_fromstring
+from odoo.tests.common import at_install, post_install, HttpCase
+from odoo.addons.mail.models import mail_template
+from odoo.tools.misc import mute_logger
+
+
+@at_install(False)
```
Comment (yajo) [inline auth_signup_verify_email/tests/test_verify_email.py]:
```suggestion
```

**Review by yajo — CHANGES_REQUESTED** [review]
I opened https://github.com/odoo/odoo/pull/34690 so please link it in the hacks, and let's see if in the future we can remove the HACK.

**Comment by MiquelRForgeFlow** [comment]
conflicts! rebase please

**Comment by Tardo** [comment]
@mreficent rebased, thanks!

**Comment by pedrobaeza** [comment]
@Yajo, please re-review

**Review by rafaelbn — APPROVED** [review]
Missing technical final review of @Yajo 

But, functionally :+1:

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Please squash the migration-related commits.

File: auth_signup_verify_email/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2015 Antiun Ingeniería, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Verify email at signup",
```
Comment (MiquelRForgeFlow) [inline auth_signup_verify_email/__manifest__.py:4]:
```suggestion
    "name": "Auth Signup Verify Email",
```

Comment (pedrobaeza) [reply auth_signup_verify_email/__manifest__.py:4]: Why? The name is correct. It mustn't be the "technical name".

**Comment by Tardo** [comment]
@Yajo done!

File: auth_signup_verify_email/__manifest__.py (hunk 9 above)
Comment (MiquelRForgeFlow) [reply auth_signup_verify_email/__manifest__.py:4]: Ok, but at least the full name could be put in uppercase.

Comment (pedrobaeza) [reply auth_signup_verify_email/__manifest__.py:4]: I don't like that uppercasing as well... and it's not mandatory in English (but tolerated).

**Pull request merged**